### PR TITLE
[FEATURE REQUEST] - Add RedirectToRoute and RedirectBack

### PR DESCRIPTION
**Pull request by sujit-baniya**
This PR creates two functions `RedirectToRoute` and `RedirectBack`.

`RedirectToRoute` - will redirect to location (along with parameters) based on route name
`RedirectBack` - will redirect back to the referer URL or the fallback defined

### CODE REVIEW

**Comment by ReneWerner87** [comment]
- [ ] Don't forget the documentation repository

**Comment by sujit-baniya** [comment]
@ReneWerner87 created PR for doc as well: https://github.com/gofiber/docs/pull/238

**Comment by ReneWerner87** [comment]
Will do the deep check tomorrow.

File: ctx.go [hunk 1]
```diff
@@ -1054,6 +1054,40 @@ func (c *Ctx) Redirect(location string, status ...int) error {
 	return nil
 }
 
+// Redirect to the URL derived from the specified path, with specified status.
+// If status is not specified, status defaults to 302 Found.
+func (c *Ctx) Redirect(location string, status ...int) error {
```
Comment (ReneWerner87) [inline ctx.go:1059]: could you explain the benefit of this wrapper function ?
we could use the Public function everywhere

File: ctx_test.go [hunk 2]
```diff
@@ -2026,6 +2026,37 @@ func Test_Ctx_Redirect(t *testing.T) {
...
+		return c.JSON(c.Params("name"))
+	}).Name("user")
+	c := app.AcquireCtx(&fasthttp.RequestCtx{})
+	defer app.ReleaseCtx(c)
+
+	c.RedirectToRoute("user", Map{
+		"name": "fiber",
+	})
+	utils.AssertEqual(t, 302, c.Response().StatusCode())
+	utils.AssertEqual(t, "/user/fiber", string(c.Response().Header.Peek(HeaderLocation)))
```
Comment (ReneWerner87) [inline ctx_test.go:2043]: can you sample some more complex examples with wildcard and optional parameters with

tetscases: wildcard, plus char(greedy-not optional), optional paramter, normal parameter

File: ctx_test.go [hunk 3]
```diff
@@ -2026,6 +2026,37 @@ func Test_Ctx_Redirect(t *testing.T) {
...
+}
+
+// go test -run Test_Ctx_RedirectBack
+func Test_Ctx_RedirectBack(t *testing.T) {
```
Comment (ReneWerner87) [inline ctx_test.go:2047]: please add a test case with referer

File: ctx_test.go (hunk 2 above)
Comment (ReneWerner87) [inline ctx_test.go:2043]: can you sample some more complex examples with wildcard and optional parameters with

tetscases: wildcard, plus char(greedy-not optional), optional paramter, normal parameter

File: ctx.go [hunk 4]
```diff
@@ -1054,6 +1054,40 @@ func (c *Ctx) Redirect(location string, status ...int) error {
...
+// RedirectToRoute to the Route registered in the app with appropriate parameters
+// If status is not specified, status defaults to 302 Found.
+func (c *Ctx) RedirectToRoute(routeName string, params Map, status ...int) error {
+	route := c.App().GetRoute(routeName)
+	location := ""
+	for _, segment := range route.routeParser.segs {
+		location = fmt.Sprintf("%s%s", location, segment.Const)
+		if segment.IsParam {
+			if val, ok := params[segment.ParamName]; ok {
+				location = fmt.Sprintf("%s%s", location, val)
```
Comment (ReneWerner87) [inline ctx.go:1072]: please add a benchmark method for this

would like to have the possibility to optimize this later on

**Review by ReneWerner87 — CHANGES_REQUESTED** [review]

File: ctx.go (hunk 1 above)
Comment (sujit-baniya) [reply ctx.go:1059]: ah yes! will update function.

File: ctx.go (hunk 4 above)
Comment (sujit-baniya) [reply ctx.go]: Done

File: ctx_test.go (hunk 3 above)
Comment (sujit-baniya) [reply ctx_test.go:2096]: Done

**Comment by sujit-baniya** [comment]
@ReneWerner87 All suggestions are implemented

File: ctx_test.go (hunk 2 above)
Comment (ReneWerner87) [reply ctx_test.go:2043]: i miss the wildcard case, but its okay, since we have the plus case

**Comment by ReneWerner87** [comment]
please fix the golint comments

File: utils/convert.go [hunk 5]
```diff
@@ -83,3 +86,55 @@ func ByteSize(bytes uint64) string {
 	result = strings.TrimSuffix(result, ".0")
 	return result + unit
 }
+
+// ToString Change arg to string
+func ToString(arg interface{}, timeFormat ...string) string {
+	if len(timeFormat) > 1 {
```
Comment (ReneWerner87) [inline utils/convert.go:92]: can you remove these unnecessary lines, when someone passes multiple formats, they are not checked or taken into account

File: utils/convert.go [hunk 6]
```diff
@@ -83,3 +86,55 @@ func ByteSize(bytes uint64) string {
...
+	case []byte:
+		return string(v)
+	case bool:
+		return strconv.FormatBool(v)
+	case float32:
+		return strconv.FormatFloat(float64(v), 'f', -1, 32)
+	case float64:
+		return strconv.FormatFloat(v, 'f', -1, 64)
+	case time.Time:
+		if len(timeFormat) == 1 {
```
Comment (ReneWerner87) [inline utils/convert.go:129]: just check for "> 0"

File: ctx.go [hunk 7]
```diff
@@ -1060,6 +1060,52 @@ func (c *Ctx) Redirect(location string, status ...int) error {
...
+// get URL location from route using parameters
+func (c *Ctx) getLocationFromRoute(route Route, params Map) (string, error) {
+	buf := bytebufferpool.Get()
+	for _, segment := range route.routeParser.segs {
+		if segment.IsParam {
+			for key, val := range params {
+				if key == segment.ParamName || segment.IsGreedy {
+					_, err := buf.WriteString(utils.ToString(val))
+					if err != nil {
+						fmt.Println(err)
```
Comment (ReneWerner87) [inline ctx.go:1072]: please remove this line

**Review by ReneWerner87 — CHANGES_REQUESTED** [review]

**Review by ReneWerner87 — APPROVED** [review]

**Pull request merged**